### ISSUE TITLE
image-to-svg: 11x pipeline speedup (MiniBatchKMeans + cv2.Sobel + C label assign)

### ISSUE THREAD

**Issue by oaustegard** [oaustegard/claude-skills#476]
## Summary

Three changes to the image-to-svg pipeline that reduce 8-variant batch time from 189s to 17s (11x speedup).

### 1. Replace `edges()` Python pixel loop with `cv2.Sobel` (edge_map: 11s → 0.13s)

The `seeing-images` `edges()` function does Sobel via `getpixel()`/`putpixel()` over every pixel. `cv2.Sobel` does the same computation in compiled C++.

In `pipeline.py`, replace the `edge_map` task:

```python
@task(depends_on=[quantize])
def edge_map(quantize):
    h, w = quantize["h"], quantize["w"]
    img = cv2.imread(_cfg["source_path"])
    gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
    gray = cv2.resize(gray, (w, h))
    gx = cv2.Sobel(gray, cv2.CV_64F, 1, 0, ksize=3)
    gy = cv2.Sobel(gray, cv2.CV_64F, 0, 1, ksize=3)
    mag = np.sqrt(gx**2 + gy**2)
    edge_img = np.clip(mag, 0, 255).astype(np.uint8)
    edge_img[edge_img <= 50] = 0
    print(f"  edge_map: {edge_img.shape}")
    return {"edge_img": edge_img}
```

This removes the dependency on the `seeing-images` skill for this step.

### 2. Replace `cv2.kmeans` with `sklearn.MiniBatchKMeans` (quantize fit: 41s → 0.6s at K=64)

`cv2.kmeans` runs 5 full restarts × 100 iterations. `MiniBatchKMeans` converges in 0.6s with visually equivalent results for this use case.

```python
from sklearn.cluster import MiniBatchKMeans
mbk = MiniBatchKMeans(n_clusters=K, batch_size=4096, max_iter=30, n_init=2, random_state=42)
mbk.fit(pixels)
centers = mbk.cluster_centers_.astype(np.uint8)
```

### 3. Replace numpy label assignment with compiled C binary (5.9s → 0.22s, 27x)

The full-resolution label assignment creates `(50000 × K × 3)` intermediate arrays per batch.
A 40-line C binary (`nn_assign.c`) does argmin nearest-neighbor in a tight loop.

The C source should live alongside the pipeline. The pipeline should compile it on first use if the binary doesn't exist, with numpy fallback if gcc isn't available.

```c
// nn_assign.c — reads pixel binary + centers binary, writes label binary to stdout
// gcc -O3 -march=native -o nn_assign nn_assign.c -lm
```

## Invariants

- SVG output must be structurally identical (same contours, same z-order) — only clustering may vary slightly due to MiniBatchKMeans stochastic initialization
- `random_state=42` ensures reproducibility across runs
- The C binary must have a numpy fallback for environments without gcc
- Shape counts should be within ~5% of original (MiniBatchKMeans produces slightly different cluster boundaries)

## Measurements (1206×1597 photo, 8 variants)

| Step | Before | After | Speedup |
|------|--------|-------|---------|
| edge_map | 11.0s | 0.13s | 85x |
| quantize (K=64) | 52.4s | 5.0s | 10x |
| label assign (K=64) | 5.9s | 0.22s | 27x |
| **Batch total** | **189.4s** | **17.1s** | **11x** |
